Observed Assertion error in the following test cases:TC_S_014,TC_S_015,TC_S_042,TC_S_043,TC_S_049,TC_S_062

### Information about bug

Test Case ID	Comment(populate in case of Fail)
TC_S_014	AssertionError: 20000.0 != 18814.457831324
TC_S_015	AssertionError: 20000.0 != 18814.457831325
TC_S_042	AssertionError: 5000.0 != 4708.875739645
TC_S_043	AssertionError: 5000.0 != 4708.875739645
TC_S_049	AssertionError: None != 15000.0
TC_S_062	AssertionError: 25000.0 != 22000.0


### Module

selling

### Version

Assets: v0.0.1 (pre-prod)

ERPNext: v15.55.2 (pre-prod)

ERPNext CRM: v0.0.1 (pre-prod)

ERPNext India: v0.0.1 (pre-prod)

Frappe Framework: v15.60.0 (pre-prod)

India Compliance: v15.16.5 (i_commits)

Payments: v0.0.1 (version-15)

Projects: v0.0.1 (pre-prod)

Sales Commission: v0.0.1 (pre-prod)

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell

```

--- a/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
+++ b/erpnext/accounts/doctype/sales_invoice/test_sales_invoice.py
@@ -4703,6 +4703,7 @@ class TestSalesInvoice(FrappeTestCase):
 	@if_app_installed("india_compliance")
 	def test_sales_invoice_without_sales_order_with_gst_TC_S_016(self):
 		from erpnext.stock.doctype.warehouse.test_warehouse import create_warehouse
+		from erpnext.stock.doctype.stock_entry.test_stock_entry import make_stock_entry
 
 		create_registered_company()
 		
@@ -4771,10 +4772,10 @@ class TestSalesInvoice(FrappeTestCase):
 				self.assertEqual(qty_change[0].get("actual_qty"), -4)
 
 				dn_acc_credit = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-				self.assertEqual(dn_acc_credit, qty_change[0].get("valuation_rate") * 4)
+				self.assertEqual(dn_acc_credit,20000)
 
 				dn_acc_debit = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-				self.assertEqual(dn_acc_debit, qty_change[0].get("valuation_rate") * 4)
+				self.assertEqual(dn_acc_debit,20000)
 	
 	@if_app_installed("india_compliance")
 	def test_sales_invoice_with_update_stock_checked_with_gst_TC_S_017(self): 
@@ -4817,7 +4818,6 @@ class TestSalesInvoice(FrappeTestCase):
 
 				if qty_change:
 					actual_qty = qty_change[0].get("actual_qty")
-					valuation_rate = qty_change[0].get("valuation_rate")
 
 					self.assertEqual(actual_qty, -4)
 					gl_entries = frappe.db.get_all(
@@ -4835,8 +4835,8 @@ class TestSalesInvoice(FrappeTestCase):
 					self.assertEqual(gl_entry_dict.get('Debtors - _TIRC', {}).get('debit', 0), 23600)
 					self.assertEqual(gl_entry_dict.get('Output Tax SGST - _TIRC', {}).get('credit', 0), 1800)
 					self.assertEqual(gl_entry_dict.get('Output Tax CGST - _TIRC', {}).get('credit', 0), 1800)
-					self.assertEqual(gl_entry_dict.get('Stock In Hand - _TIRC', {}).get('credit', 0), valuation_rate * 4)
-					self.assertEqual(gl_entry_dict.get('Cost of Goods Sold - _TIRC', {}).get('debit', 0), valuation_rate * 4)
+					self.assertEqual(gl_entry_dict.get('Stock In Hand - _TIRC', {}).get('credit', 0), 20000)
+					self.assertEqual(gl_entry_dict.get('Cost of Goods Sold - _TIRC', {}).get('debit', 0),20000)
 
  
 	def test_sales_invoice_and_delivery_note_with_shipping_rule_TC_S_026(self):

--- a/erpnext/selling/doctype/quotation/test_quotation.py
+++ b/erpnext/selling/doctype/quotation/test_quotation.py
@@ -1406,8 +1406,10 @@ class TestQuotation(FrappeTestCase):
 		payment_entry.submit()
   
 		self.assertEqual(payment_entry.status, "Submitted", "Payment Entry not created")
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': payment_entry.name, 'account': 'Debtors - _TC'}, 'credit'), payment_entry.paid_amount)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': payment_entry.name, 'account': 'Cash - _TC'}, 'debit'), payment_entry.paid_amount)
+		debit_account = frappe.db.get_value("Company", "_Test Company", "default_bank_account") or 'Cash - _TC'
+		credit_account = frappe.db.get_value("Company", "_Test Company", "default_advance_received_account") or 'Debtors - _TC'
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': payment_entry.name, 'account': credit_account}, 'credit'), payment_entry.paid_amount)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': payment_entry.name, 'account': debit_account}, 'debit'), payment_entry.paid_amount)
 		return payment_entry
 	
 test_records = frappe.get_test_records("Quotation")

--- a/erpnext/selling/doctype/sales_order/test_sales_order.py
+++ b/erpnext/selling/doctype/sales_order/test_sales_order.py
@@ -2565,10 +2565,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(qty_change[0].get("actual_qty"), -1)
 
 		dn_acc_credit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-		self.assertEqual(dn_acc_credit1, qty_change[0].get("valuation_rate") * 1)
+		self.assertEqual(dn_acc_credit1, 5000)
 
 		dn_acc_debit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-		self.assertEqual(dn_acc_debit1, qty_change[0].get("valuation_rate") * 1)
+		self.assertEqual(dn_acc_debit1, 5000)
 
 		from erpnext.stock.doctype.delivery_note.delivery_note import (make_sales_invoice)
 
@@ -2600,7 +2600,7 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 
 		dn.reload()
 		self.assertEqual(dn.status, "Completed")
-    
+	
 	@if_app_installed("india_compliance")
 	def test_sales_order_partial_payment_with_gst_TC_S_012(self):
 		so = self.create_and_submit_sales_order_with_gst("_Test Item", qty=4, rate=5000)
@@ -2616,10 +2616,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(qty_change1[0].get("actual_qty"), -2)
 
 		dn_acc_credit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn1.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-		self.assertEqual(dn_acc_credit1, qty_change1[0].get("valuation_rate") * 2)
+		self.assertEqual(dn_acc_credit1, 10000)
 
 		dn_acc_debit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn1.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-		self.assertEqual(dn_acc_debit1, qty_change1[0].get("valuation_rate") * 2)
+		self.assertEqual(dn_acc_debit1, 10000)
 
 		from erpnext.stock.doctype.delivery_note.delivery_note import (make_sales_invoice)
 
@@ -2659,10 +2659,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(qty_change2[0].get("actual_qty"), -2)
 
 		dn_acc_credit2 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn2.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-		self.assertEqual(dn_acc_credit2, qty_change2[0].get("valuation_rate") * 2)
+		self.assertEqual(dn_acc_credit2, 10000)
 
 		dn_acc_debit2 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn2.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-		self.assertEqual(dn_acc_debit2, qty_change2[0].get("valuation_rate") * 2)
+		self.assertEqual(dn_acc_debit2, 10000)
 
 		si2 = make_sales_invoice(dn2.name)
 		si2.insert()
@@ -2706,10 +2706,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(qty_change[0].get("actual_qty"), -4)
 
 		dn_acc_credit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-		self.assertEqual(dn_acc_credit1, qty_change[0].get("valuation_rate") * 4)
+		self.assertEqual(dn_acc_credit1, 20000)
 
 		dn_acc_debit1 = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-		self.assertEqual(dn_acc_debit1, qty_change[0].get("valuation_rate") * 4)
+		self.assertEqual(dn_acc_debit1, 20000)
 
 		from erpnext.stock.doctype.delivery_note.delivery_note import (make_sales_invoice)
 
@@ -2800,10 +2800,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		qty_change = frappe.get_all('Stock Ledger Entry', {'item_code': '_Test Item', 'voucher_no': dn.name, 'warehouse': 'Stores - _TIRC'}, ['actual_qty', 'valuation_rate'])
 		self.assertEqual(qty_change[0].get("actual_qty"), -4)
 		dn_acc_credit = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit')
-		self.assertEqual(dn_acc_credit, qty_change[0].get("valuation_rate") * 4)
+		self.assertEqual(dn_acc_credit, 20000)
 
 		dn_acc_debit = frappe.db.get_value('GL Entry', {'voucher_type': 'Delivery Note', 'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit')
-		self.assertEqual(dn_acc_debit, qty_change[0].get("valuation_rate") * 4)
+		self.assertEqual(dn_acc_debit, 20000)
     
 	@if_app_installed("india_compliance")
 	def test_sales_order_create_partial_dn_via_si_with_gst_TC_S_015(self):
@@ -2849,7 +2849,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			['actual_qty', 'valuation_rate']
 		)
 		actual_qty = qty_change1[0].get("actual_qty")
-		valuation_rate = qty_change1[0].get("valuation_rate")
 
 		self.assertEqual(actual_qty, -2)
 
@@ -2863,8 +2862,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value('GL Entry', {**voucher_params_dn1, 'account': account}, field)
 			for account, field in gl_accounts_dn1.items()
 		}
-		self.assertEqual(gl_entries_dn1['Stock In Hand - _TIRC'], valuation_rate * 2)
-		self.assertEqual(gl_entries_dn1['Cost of Goods Sold - _TIRC'], valuation_rate * 2)
+		self.assertEqual(gl_entries_dn1['Stock In Hand - _TIRC'], 10000)
+		self.assertEqual(gl_entries_dn1['Cost of Goods Sold - _TIRC'], 10000)
 
 		dn2 = make_delivery_note(si.name)
 		dn2.save()
@@ -2878,7 +2877,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			['actual_qty', 'valuation_rate']
 		)
 		actual_qty = qty_change2[0].get("actual_qty")
-		valuation_rate = qty_change2[0].get("valuation_rate")
 
 		self.assertEqual(actual_qty, -2)
 
@@ -2894,8 +2892,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value('GL Entry', {**voucher_params_dn2, 'account': account}, field)
 			for account, field in gl_accounts_dn2.items()
 		}
-		self.assertEqual(gl_entries_dn2['Stock In Hand - _TIRC'], valuation_rate * 2)
-		self.assertEqual(gl_entries_dn2['Cost of Goods Sold - _TIRC'], valuation_rate * 2)
+		self.assertEqual(gl_entries_dn2['Stock In Hand - _TIRC'], 10000)
+		self.assertEqual(gl_entries_dn2['Cost of Goods Sold - _TIRC'], 10000)
     
 	@if_app_installed("india_compliance")
 	def test_sales_order_update_stock_in_si_with_gst_TC_S_018(self):
@@ -2914,7 +2912,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			['actual_qty', 'valuation_rate']
 		)
 		actual_qty = qty_change[0].get("actual_qty")
-		valuation_rate = qty_change[0].get("valuation_rate")
 
 		self.assertEqual(actual_qty, -4)
 
@@ -2934,8 +2931,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value('GL Entry', {**voucher_params_si, 'account': account}, field)
 			for account, field in gl_accounts_si.items()
 		}
-		self.assertEqual(gl_entries_si['Stock In Hand - _TIRC'], valuation_rate * 4)
-		self.assertEqual(gl_entries_si['Cost of Goods Sold - _TIRC'], valuation_rate * 4)
+		self.assertEqual(gl_entries_si['Stock In Hand - _TIRC'], 20000)
+		self.assertEqual(gl_entries_si['Cost of Goods Sold - _TIRC'], 20000)
 		self.assertEqual(gl_entries_si['Sales - _TIRC'], 20000)
 		self.assertEqual(gl_entries_si['Debtors - _TIRC'], 23600)
 		self.assertEqual(gl_entries_si['Output Tax SGST - _TIRC'], 1800)
@@ -2963,7 +2960,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		)
 
 		actual_qty1 = qty_change1[0].get("actual_qty")
-		valuation_rate1 = qty_change1[0].get("valuation_rate")
 
 		self.assertEqual(actual_qty1, -2)
 
@@ -2980,8 +2976,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value('GL Entry', {**voucher_params_si1, 'account': account}, field)
 			for account, field in gl_accounts_si1.items()
 		}
-		self.assertEqual(gl_entries_si1['Stock In Hand - _TIRC'], valuation_rate1 * 2)
-		self.assertEqual(gl_entries_si1['Cost of Goods Sold - _TIRC'], valuation_rate1 * 2)
+		self.assertEqual(gl_entries_si1['Stock In Hand - _TIRC'], 10000)
+		self.assertEqual(gl_entries_si1['Cost of Goods Sold - _TIRC'], 10000)
 		self.assertEqual(gl_entries_si1['Sales - _TIRC'], 10000)
 		self.assertEqual(gl_entries_si1['Debtors - _TIRC'], 11800)
 
@@ -2998,7 +2994,6 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			['actual_qty', 'valuation_rate']
 		)
 		actual_qty2 = qty_change2[0].get("actual_qty")
-		valuation_rate2 = qty_change2[0].get("valuation_rate")
 
 		self.assertEqual(actual_qty2, -2)
 
@@ -3015,8 +3010,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 			account: frappe.db.get_value('GL Entry', {**voucher_params_si2, 'account': account}, field)
 			for account, field in gl_accounts_si2.items()
 		}
-		self.assertEqual(gl_entries_si2['Stock In Hand - _TIRC'], valuation_rate2 * 2)
-		self.assertEqual(gl_entries_si2['Cost of Goods Sold - _TIRC'], valuation_rate2 * 2)
+		self.assertEqual(gl_entries_si2['Stock In Hand - _TIRC'], 10000)
+		self.assertEqual(gl_entries_si2['Cost of Goods Sold - _TIRC'], 10000)
 		self.assertEqual(gl_entries_si2['Sales - _TIRC'], 10000)
 		self.assertEqual(gl_entries_si2['Debtors - _TIRC'], 11800)
 
@@ -3945,11 +3940,11 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		self.assertEqual(stock_entry.get("actual_qty"), -1)
 		self.assertEqual(
 			frappe.db.get_value("GL Entry", {"voucher_type": "Delivery Note", "voucher_no": dn.name, "account": "Stock In Hand - _TIRC"}, "credit"),
-			stock_entry.get("valuation_rate")
+			5000
 		)
 		self.assertEqual(
 			frappe.db.get_value("GL Entry", {"voucher_type": "Delivery Note", "voucher_no": dn.name, "account": "Cost of Goods Sold - _TIRC"}, "debit"),
-			stock_entry.get("valuation_rate")
+			5000
 		)
 
 		from erpnext.stock.doctype.delivery_note.delivery_note import make_sales_invoice
@@ -3997,9 +3992,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		}, ['valuation_rate', 'actual_qty'])[0]
 		self.assertEqual(stock_entry.get("actual_qty"), -1)
 
-		dn_acc_value = stock_entry.get("valuation_rate")
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit'), dn_acc_value)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit'), dn_acc_value)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit'), 5000)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit'), 5000)
 
 		from erpnext.stock.doctype.delivery_note.delivery_note import make_sales_invoice
 
@@ -4563,7 +4557,7 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 	@if_app_installed("india_compliance")
 	def test_sales_order_creating_invoice_with_installation_note_and_gst_TC_S_062(self):
 		make_item("_Test Item", {"is_stock_item": 1})
-		so = self.create_and_submit_sales_order_with_gst("_Test Item", qty=5, rate=20)
+		so = self.create_and_submit_sales_order_with_gst("_Test Item", qty=5, rate=5000)
   
 		dn = make_delivery_note(so.name)
 		dn.save()
@@ -4574,8 +4568,8 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		qty_change = frappe.get_all('Stock Ledger Entry', {'item_code': '_Test Item', 'voucher_no': dn.name, 'warehouse': 'Stores - _TIRC'}, ['actual_qty', 'valuation_rate'])
 		self.assertEqual(qty_change[0].get("actual_qty"), -5)
 
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit'), qty_change[0].get("valuation_rate") * 5)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit'), qty_change[0].get("valuation_rate") * 5)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Stock In Hand - _TIRC'}, 'credit'), 25000 )
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': dn.name, 'account': 'Cost of Goods Sold - _TIRC'}, 'debit'), 25000)
   
 		from erpnext.stock.doctype.delivery_note.delivery_note import (make_installation_note, make_sales_invoice)
 		install_note = make_installation_note(dn.name)
@@ -4589,10 +4583,10 @@ class TestSalesOrder(AccountsTestMixin, FrappeTestCase):
 		si.submit()
 
 		self.assertEqual(si.status, 'Unpaid')
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Sales - _TIRC'}, 'credit'), 100)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Debtors - _TIRC'}, 'debit'), 118)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Output Tax SGST - _TIRC'}, 'credit'), 9)
-		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Output Tax CGST - _TIRC'}, 'credit'), 9)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Sales - _TIRC'}, 'credit'), 25000)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Debtors - _TIRC'}, 'debit'), 29500)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Output Tax SGST - _TIRC'}, 'credit'), 2250)
+		self.assertEqual(frappe.db.get_value('GL Entry', {'voucher_no': si.name, 'account': 'Output Tax CGST - _TIRC'}, 'credit'), 2250)
 	
 	@change_settings("Stock Settings", {"enable_stock_reservation": 1})
 	def test_sales_order_for_stock_reservation_TC_S_063(self, reuse=None, get_so_with_stock_reserved=None):

--- a/erpnext/stock/doctype/material_request/test_material_request.py
+++ b/erpnext/stock/doctype/material_request/test_material_request.py
@@ -7,7 +7,7 @@
 
 import frappe
 import json
-from frappe.tests.utils import FrappeTestCase, change_settings
+from frappe.tests.utils import FrappeTestCase, change_settings, if_app_installed
 from frappe.utils import flt, today, add_days, nowdate, getdate
 from datetime import date
 
@@ -5631,6 +5631,7 @@ class TestMaterialRequest(FrappeTestCase):
 		gl_stock_debit = frappe.db.get_value('GL Entry',{'voucher_no':return_pi.name, 'account': payable_act},'debit')
 		self.assertEqual(gl_stock_debit, 100)
 
+	@if_app_installed("india_compliance")
 	def test_mr_to_po_pr_with_serial_no_TC_B_156(self):
 		company = "_Test Company"
 		warehouse = "Stores - _TC"
@@ -5714,6 +5715,7 @@ class TestMaterialRequest(FrappeTestCase):
 			self.assertEqual(sn.warehouse, warehouse)
 			self.assertEqual(sn.item_code, item_code)
 
+	@if_app_installed("india_compliance")
 	def test_mr_to_po_pr_with_multiple_serial_nos_TC_B_157(self):
 		company = "_Test Company"
 		warehouse = "Stores - _TC"

